### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by wernermayer** [owncloud/music#351]
Hello, thanks for this great app,
but when i try to upload something into a link shared folder,  the file gets uploaded correctly but i get an internal server error and the shown content of the folder don't becomes updated so i have to reload the page with f5.
 In the ownCloud logs there is an
 "Call to a member function get() on a non-object at /var/www/owncloud/apps/music/utility/scanner.php#85" Error.
After log in when i use the nomal file upload system it works fine, 
but link shared folder upload only works correctly with deactivated music app.
Is this a bug or what can i do?
I Have Ubuntu 14.04, Apache, PHP-FPM, mariadb, appversion 0.3.5, and ownCloud 8.0.0.7

**Comment by MorrisJobke** [comment]
@wernermayer Thanks for the report. I need to have a look at the code for the shared folders.

**Comment by katosh** [comment]
I can confirm this exact error. Also when the music app is scanning for new files while this error occurs the scanning process freezes. I downloaded ownclouds logfile but it turned into a mess so here is only the last entry:
```
Exception: {"Message":"An exception occured in driver: SQLSTATE[HY000] [2003] Can't connect to MySQL server on '127.0.0.1' (4)","Code":0,"Trace":"#0 \/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Connection.php(429): OC\\DB\\Connection->connect()\n#1 \/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Connection.php(389): Doctrine\\DBAL\\Connection->getDatabasePlatformVersion()\n#2 \/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Connection.php(328): Doctrine\\DBAL\\Connection->detectDatabasePlatform()\n#3 \/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Connection.php(621): Doctrine\\DBAL\\Connection->getDatabasePlatform()\n#4 \/var\/www\/owncloud\/lib\/private\/db\/connection.php(66): Doctrine\\DBAL\\Connection->setTransactionIsolation(2)\n#5 \/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/DriverManager.php(172): OC\\DB\\Connection->__construct(Array, Object(Doctrine\\DBAL\\Driver\\PDOMySql\\Driver), Object(Doctrine\\DBAL\\Configuration), Object(Doctrine\\Common\\EventManager))\n#6 \/var\/www\/owncloud\/lib\/private\/db\/connectionfactory.php(107): Doctrine\\DBAL\\DriverManager::getConnection(Array, Object(Doctrine\\DBAL\\Configuration), Object(Doctrine\\Common\\EventManager))\n#7 \/var\/www\/owncloud\/lib\/private\/server.php(243): OC\\DB\\ConnectionFactory->getConnection('mysql', Array)\n#8 \/var\/www\/owncloud\/3rdparty\/pimple\/pimple\/src\/Pimple\/Container.php(112): OC\\Server->OC\\{closure}(Object(OC\\Server))\n#9 \/var\/www\/owncloud\/lib\/private\/appframework\/utility\/simplecontainer.php(91): Pimple\\Container->offsetGet('DatabaseConnect...')\n#10 \/var\/www\/owncloud\/lib\/private\/server.php(540): OC\\AppFramework\\Utility\\SimpleContainer->query('DatabaseConnect...')\n#11 \/var\/www\/owncloud\/lib\/private\/db.php(33): OC\\Server->getDatabaseConnection()\n#12 \/var\/www\/owncloud\/lib\/private\/server.php(178): OC_DB::getConnection()\n#13 \/var\/www\/owncloud\/3rdparty\/pimple\/pimple\/src\/Pimple\/Container.php(112): OC\\Server->OC\\{closure}(Object(OC\\Server))\n#14 \/var\/www\/owncloud\/lib\/private\/appframework\/utility\/simplecontainer.php(91): Pimple\\Container->offsetGet('AppConfig')\n#15 \/var\/www\/owncloud\/lib\/private\/server.php(488): OC\\AppFramework\\Utility\\SimpleContainer->query('AppConfig')\n#16 \/var\/www\/owncloud\/lib\/private\/app.php(218): OC\\Server->getAppConfig()\n#17 \/var\/www\/owncloud\/lib\/private\/app.php(71): OC_App::getEnabledApps()\n#18 \/var\/www\/owncloud\/lib\/base.php(542): OC_App::loadApps(Array)\n#19 \/var\/www\/owncloud\/lib\/base.php(1026): OC::init()\n#20 \/var\/www\/owncloud\/index.php(34): require_once('\/var\/www\/ownclo...')\n#21 {main}","File":"\/var\/www\/owncloud\/lib\/private\/db\/connection.php","Line":33}
```

**Comment by paulijar** [comment]
The actual error reported here seems to be fixed a long time ago, but uploading music to a public share still did not update the Music DB automatically in the latest release 0.4.3. A warning was printed to log on such audio upload. This automatic DB updating is now fixed with commit ee48c2612d58.
